h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -334,30 +327,41 @@ public void computeImpl() {
             throw new IllegalArgumentException("Found validation errors: " + validationErrors());
           }
 
-          // Compute SVD of Gram A'A/n using JAMA library
-          // Note: Singular values ordered in weakly descending order by algorithm
+          // Compute SVD of Gram A'A/n using netlib-java (MTJ) library
           _job.update(1, "Calculating SVD of Gram matrix locally");
-          Matrix gramJ = _wideDataset ? new Matrix(ogtsk._gram.getXX()) : new Matrix(gtsk._gram.getXX());
-          SingularValueDecomposition svdJ = gramJ.svd();
-          _job.update(1, "Computing stats from SVD");
-          // correct for the eigenvector by t(A)*eigenvector for wide dataset
-          if (_wideDataset) {
-            double[][] eigenVecs = transformEigenVectors(dinfo, svdJ.getV().getArray());
-            computeStatsFillModel(model, dinfo, svdJ.getSingularValues(), eigenVecs, gram, model._output._nobs);
-          } else {
-            computeStatsFillModel(model, dinfo, svdJ, gram, model._output._nobs);
+          double[][] gramMatrix;
+          try {
+            gramMatrix = _wideDataset ? ogtsk._gram.getXX() : gtsk._gram.getXX();
+          } catch (NullPointerException e) {
```

Comment:
Why dedicated NPE check is here?

Reply:
If it is debugging construct, then remove it

Reply:
OK, I have removed it.